pkg/decision/composite_feature_service.go

```
@@ -51,6 +53,21 @@ func (f CompositeFeatureService) GetDecision(decisionContext FeatureDecisionCont
 		reasons.Append(decisionReasons)
 		if err != nil {
 			f.logger.Debug(err.Error())
+			// Check if this is a CMAB error - if so, stop the loop and return empty decision
+			if strings.Contains(err.Error(), "Failed to fetch CMAB data") {
+				// Add the CMAB error to reasons before returning - use AddError for critical failures
+				reasons.AddError(err.Error())
+				return FeatureDecision{}, reasons, nil // Return empty decision for CMAB errors
```

it should return the error here